server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -630,6 +634,12 @@ private void balanceByWeights(NodeSorter sorter) {
                             sorter.sort(0, relevantNodes);
                             lowIdx = 0;
                             highIdx = relevantNodes - 1;
+                            if (allocation.hasTargetRelocatingShards()) {
+                                // The shard may have been relocated on the ModelNode, but not on the cluster (RoutingNodes) if
+                                // it is throttled. The `hasTargetRelocatingShards` check does not account for this case.
+                                // This might be a non-issue since our intention is to have no Throttle with early returns.
+                                return;
```

Comment:
Could we change balanceByWeights to return a boolean, instead of void, to indicate whether something was moved? Then anyone adding a return statement to this method in future must consider the answer: less fragile.

Reply:
Yeah this was temporary. It now returns boolean see dbd0fede791b9a31aa750108d32f7d9b99775d18